src/hotspot/share/gc/g1/g1CardSetContainers.hpp

```diff
@@ -238,23 +238,27 @@ class G1CardSetHowl : public G1CardSetContainer {
   using ContainerPtr = G1CardSet::ContainerPtr;
   EntryCountType volatile _num_entries;
 private:
-  ContainerPtr _buckets[2];
-  // Do not add class member variables beyond this point
+  // VLA implementation.
+  ContainerPtr _buckets[1];
+  // Do not add class member variables beyond this point.
 
   // Iterates over the given ContainerPtr with at index in this Howl card set,
   // applying a CardOrRangeVisitor on it.
   template <class CardOrRangeVisitor>
   void iterate_cardset(ContainerPtr const container, uint index, CardOrRangeVisitor& found, G1CardSetConfiguration* config);
 
+  ContainerPtr at(EntryCountType index);
```

Comment:
Could be made `const`, i.e. 
```suggestion
  ContainerPtr at(EntryCountType index) const;
```